pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,143 @@
 """Tests that sync container plugin repositories."""
-from pulp_smash import config
-from pulp_smash.pulp3.bindings import (
-    delete_orphans,
-    monitor_task,
-    PulpTaskError,
-    PulpTestCase,
-)
-from pulp_smash.pulp3.utils import gen_repo
-from pulp_container.tests.functional.utils import (
-    gen_container_client,
-    gen_container_remote,
-)
+import pytest
+from pulpcore.tests.functional import PulpTaskError
+
+from pulp_container.tests.functional.utils import gen_container_remote
 from pulp_container.tests.functional.constants import PULP_FIXTURE_1
 
-from pulpcore.client.pulp_container import (
-    ContainerContainerRepository,
-    ContainerContainerRemote,
-    ContainerRepositorySyncURL,
-    ContentTagsApi,
-    RepositoriesContainerApi,
-    RepositoriesContainerVersionsApi,
-    RemotesContainerApi,
-)
+from pulpcore.client.pulp_container import ContainerContainerRemote
 
 
-class BasicSyncTestCase(PulpTestCase):
-    """Sync repositories with the container plugin."""
+@pytest.fixture
+def synced_repository_with_filtered_tags(container_repo, container_remote_api, sync):
+    def _synced_repository_with_filtered_tags(include_tags=None, exclude_tags=None):
+        """Sync a new repository with the included tags passed as an argument."""
+        remote_data = gen_container_remote(
+            upstream_name=PULP_FIXTURE_1,
+            include_tags=include_tags,
+            exclude_tags=exclude_tags,
+        )
+        remote = container_remote_api.create(remote_data)
```

Comment:
Do not forget to handle the removal of the remote too. You can use the `gen_object_with_cleanup` fixture for that.